### PR TITLE
Reformat build guide and add note about building serially when debugging the build process

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

None.

### Summary of the issue:

The build guide currently suggests that developers use the `--all-cores` parameter to SCons to speed up their builds, noting that this will scramble the output. However, it does not mention that in some cases this may break building altogether.

### Description of user facing changes

None. Developer change only.

### Description of development approach

Updated `projectDocs/dev/buildingNVDA.md`:

* Fixed a bunch of linting issues, including having blanks around headings, using 2 spaces for list indents, and not having duplicate headings.
* Added a heading for building NVDA without archiving it.
* Noted that the `-j N` option can be used instead of the `--all-cores` option.
* Noted that using a parallel build may break the build system.
* Noted that when experiencing problems with the build system, make sure to run it serially to make finding the source of the issue easier.

### Testing strategy:

Read over what I had written. Ran it through a spell checker and Markdown linter.

### Known issues with pull request:

None.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **Documentation**
	- Enhanced clarity and detail in the NVDA build process documentation.
	- Updated section titles for better understanding, including renaming "Building NVDA" to "Making Binary Builds."
	- Added a new subsection for creating non-archived binary builds.
	- Expanded instructions for optimizing `scons` calls with the `-j N` option.
	- Included cautionary notes and recommendations for users encountering build errors.

### CODE REVIEW

File: projectDocs/dev/buildingNVDA.md [hunk 1]
```diff
@@ -18,24 +20,31 @@ scons source user_docs
...
+
 ## Running the Source Code
+
 It is possible to run NVDA directly from source without having to build the full binary package and launcher.
 To launch NVDA from source, using `cmd.exe`, execute `runnvda.bat` in the root of the repository.
 
 To view help on the arguments that NVDA will accept, use the `-h` or `--help` option.
 These arguments are also documented in the [user guide](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#CommandLineOptions).
 
-## Building NVDA
```
Comment (seanbudd) [inline projectDocs/dev/buildingNVDA.md:34]: is this anchor used anywhere?

File: projectDocs/dev/buildingNVDA.md [hunk 2]
```diff
@@ -18,24 +20,31 @@ scons source user_docs
 
 While simply testing or committing changes, it may be faster usually just doing `scons source` as user documentation will change each time the revision number changes.
 
-You can speed up scons calls by appending the `--all-cores` parameter to the call, however note that output will be scrambled:
+You can speed up scons calls by appending the `--all-cores` or `-j N` (where `N` is the number of cores to use) parameter to the call, however note that this can cause errors, and output will be scrambled:
```
Comment (seanbudd) [inline projectDocs/dev/buildingNVDA.md:23]:
```suggestion
You can speed up scons calls by appending the following CLI parameters:

- `-j N`, where `N` is the number of cores to use while building
- `--all-cores` to use all cores.

However note that building across cores can cause errors, and output will be scrambled.
```

File: projectDocs/dev/buildingNVDA.md [hunk 3]
```diff
@@ -18,24 +20,31 @@ scons source user_docs
...
 
 ```cmd
 scons source --all-cores
```
Comment (seanbudd) [inline projectDocs/dev/buildingNVDA.md:26]:
```suggestion
scons source --all-cores
scons checkPot -j 1
```

File: projectDocs/dev/buildingNVDA.md (hunk 1 above)
Comment (SaschaCowley) [reply projectDocs/dev/buildingNVDA.md:34]: I searched the repo for `buildingNVDA.md#building-nvda` and didn't come up with anything. Note that its implicit slug would clash with the top heading though anyway.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**